pallets/index/src/lib.rs

```diff
@@ -0,0 +1,512 @@
+pub mod pallet {
+	use assets_registry::GetAssetRegistryInfo;
+	use codec::Encode;
+	use frame_support::{
+		dispatch::DispatchResult, pallet_prelude::*, traits::StorageVersion, transactional,
+		PalletId,
+	};
+	use frame_system::pallet_prelude::*;
+	use sp_runtime::traits::AccountIdConversion;
+	use sp_std::{vec, vec::Vec};
+	use xcm::latest::{prelude::*, AssetId as XcmAssetId};
+	use xcm_executor::traits::TransactAsset;
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	#[pallet::storage_version(STORAGE_VERSION)]
+	#[pallet::without_storage_info]
+	pub struct Pallet<T>(_);
+
+	pub type TaskId = [u8; 32];
+	pub const MODULE_ID: PalletId = PalletId(*b"index/ac");
+
+	#[derive(Clone, Decode, Encode, Eq, PartialEq, Ord, PartialOrd, Debug, TypeInfo)]
+	pub struct DepositInfo<AccountId> {
+		/// Sender address on current chain
+		pub sender: AccountId,
+		/// Deposit asset
+		pub asset: XcmAssetId,
+		/// Deposit amount
+		pub amount: u128,
+		/// Recipient address on dest chain
+		pub recipient: Vec<u8>,
+		/// Encoded execution plan produced by Solver
+		pub task: Vec<u8>,
+	}
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config + pallet_assets::Config {
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+		type CommitteeOrigin: EnsureOrigin<Self::RuntimeOrigin>;
+		/// Asset adapter to do withdraw, deposit etc.
+		type AssetTransactor: TransactAsset;
+		/// Assets registry
+		type AssetsRegistry: GetAssetRegistryInfo<<Self as pallet_assets::Config>::AssetId>;
+	}
+
+	const STORAGE_VERSION: StorageVersion = StorageVersion::new(0);
+
+	/// Pre-set index worker account
+	#[pallet::storage]
+	#[pallet::getter(fn executors)]
+	pub type Workers<T: Config> = StorageMap<_, Twox64Concat, T::AccountId, bool, ValueQuery>;
+
+	/// Mapping task_id to the full deposit data
+	#[pallet::storage]
+	#[pallet::getter(fn deposit_records)]
+	pub type DepositRecords<T: Config> =
+		StorageMap<_, Twox64Concat, TaskId, DepositInfo<T::AccountId>>;
+
+	/// Mapping the worker account and its actived task queue
+	#[pallet::storage]
+	#[pallet::getter(fn actived_tasks)]
+	pub type ActivedTasks<T: Config> =
+		StorageMap<_, Twox64Concat, T::AccountId, Vec<TaskId>, ValueQuery>;
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// Worker is set.
+		WorkerAdd { worker: T::AccountId },
+		/// Worker is set.
+		WorkerRemove { worker: T::AccountId },
+		/// New task saved.
+		NewTask {
+			/// Record
+			deposit_info: DepositInfo<T::AccountId>,
+		},
+		/// Task has been claimed.
+		Claimed { tasks: Vec<TaskId> },
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		AssetNotFound,
+		WorkerAlreadySet,
+		WorkerNotSet,
+		WorkerMismatch,
+		TaskAlreadyExist,
+		NotFoundInTaskQueue,
+		TaskQueueEmpty,
+		TransactFailed,
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T>
+	where
+		T::AccountId: Into<[u8; 32]> + From<[u8; 32]>,
```

Comment:
Is it still required?

Reply:
No, still need to have this constraint so that can we can convert T::AccountId to/from [u8; 32] in `do_asset_transact`